src/utilities/localization-utils.ts

```diff
@@ -0,0 +1,162 @@
+
+const defaultCultureCode = () => Rfc4646LanguageCodes.EN_US;
+
+/**
+ * Detect current language for which to provide translations
+ * @param requestCulture Incoming requested culture
+ * @returns string current RFC-4646 culture code
+ */
+const detectCultureCode = () => {
+    let culture = cultureCodeFromQueryString();
+
+    // If requested culture is missing, default to english
+    if (StringUtils.isEmpty(culture)) {
+        culture = defaultCultureCode();
+    }
+
+    if (culture === currentCultureCode()) {
+        return culture;
+    }
+
+    changeCultureCode(culture!);
+
+    return culture;
+};
+
+/**
+ * Initialize frontend i18n module - typically in root/startup of application
+ * @param module Third party module for use with i18next (ie. initReactI18next)
+ * @param cultures List of supported language cultures
+ * @param escapeValue Optional flag to set interpolation value escaping. False by default being react does this by default
+ */
+const initialize = <TResources>(
+    module: any,
+    cultures: Culture<TResources>[],
+    escapeValue: boolean = false
+) => {
+    if (CollectionUtils.isEmpty(cultures)) {
+        throw new Error(errorCultureIsRequired);
+    }
+
+    i18n.use(module).init({
+        cleanCode: true, // language will be lowercased EN --> en while leaving full locales like en-US
+        debug: EnvironmentUtils.isDevelopment(), // logs info level to console output. Helps finding issues with loading not working.
+        // detection: detectionOptions, // TODO: Issue #12 - Uncomment and implement when enabling translation by way of routes (ie. /en-us/xyz)
+        fallbackLng: defaultCultureCode(), // language to use if translations in user language are not available.
+        interpolation: {
+            escapeValue,
+        },
+        lng: defaultCultureCode(),
+        resources: culturesToResources<TResources>(cultures),
+    });
+
+    return i18n;
+};
+
+/**
+ * Retrieve translation for given key in the currently configured language
+ * @param key culture resource key
+ * @param options object key/values for interpolation of dynamic values
+ */
+const translate = (key: string, options?: any): string => i18n.t(key, options);
+
+/**
+ * Retrieve translation for given key in the currently configured language
+ * @param key culture resource key
+ * @param options object key/values for interpolation of dynamic values
+ */
+const t = translate;
+
+//#endregion Functions
+
+// -----------------------------------------------------------------------------------------
+// #region Exports
+// -----------------------------------------------------------------------------------------
+
+export const LocalizationUtils = {
+    changeCultureCode,
+    cultureCodeFromQueryString,
+    cultureFactory,
+    currentCultureCode,
+    defaultCultureCode,
+    detectCultureCode,
+    errorCultureIsRequired,
+    initialize,
+    routeParam,
+    t,
+    translate,
+};
+
+//#endregion Exports
```

Comment:
Minor, but I think your region snippet is off by a space! We usually add a space before starting comments, regions or not.